src/legacy/server/saved_objects/service/saved_objects_client.ts

```diff
@@ -99,7 +99,14 @@ export interface MigrationVersion {
 }
 
 export interface SavedObjectAttributes {
```

Comment:
This doesn't seem to allow arrays of string|number|boolean as values; not sure if that matters, but is more specific than the previous `Record<string, any>` in use elsewhere.  Curious about `undefined` as well, since that's not JSONable (a field with an `undefined` value is "deleted" during `JSON.stringify()`.

Reply:
Yeah, in regards to the arrays, the root attributes must be an object but anything nested inside can be an array. So we have `SavedObjectAttributes[]` on line 104 to allow that. In regards to `undefined`, it's to allow `myValue?: string` in TypeScript, otherwise TypeScript isn't happy and complains that values must be defined and one of `SavedObjectAttributes`.

Reply:
Something still seems not right to me, or I'm misunderstanding TS.

The following seem like objects that satisfy SavedObjectAttributes:

```
{}
{ a: 1 }
{ a: { b: 2 } }
{ a: [ { b: 2 }, { c: 3 } ] }
```

I think the following object is not a valid SavedObjectAttributes:

```
{ a: [1, 2, 3] }
```

Basically, you can only have arrays of objects, nothing else.

Reply:
I see what you mean now and you're right, I will add array support for the types that are there.